src/app/shifts/add-edit-shift/add-edit-shift.component.ts

```diff
@@ -31,38 +31,66 @@ export class AddEditShiftComponent {
     this.addingShift = this.shift() === undefined;
 
     if (!this.addingShift) {
+      //TODO: shift.date is not coming through correctly.
+      //need to figure out a way to receive dates automatically from the API
+      const dateValue = new Date(this.shift()!.date);
+
       // we're editing an existing shift; so, set inputs
       this.creditTipsInput = this.shift()!.creditTips;
       this.cashTipsInput = this.shift()!.cashTips;
       this.tipoutInput = this.shift()!.tipout;
-      this.dateInput = this.shift()!.date;
+      this.dateInput = dateValue.toISOString().slice(0, 10);
       this.hoursWorkedInput = this.shift()!.hoursWorked;
     }
   }
 
   onCancel() {
-    this.close.emit();
+    this.close.emit(undefined);
   }
 
   onSubmit() {
+    let shift: GetShiftDto | undefined = undefined;
```

Comment:
this logic should be separated out into two different components. One for adding and one for editing. this will be addressed in a future ticket

Reply:
https://nathan-tino.atlassian.net/browse/TB-15?atlOrigin=eyJpIjoiMjNiMmE4ZDc2NTJjNGJmZTgxMzJiODYzZTMzYWJlYzMiLCJwIjoiaiJ9